packages/preview-middleware-client/src/adp/controllers/AddSubpage.controller.ts

```diff
@@ -120,46 +127,60 @@ export default class AddSubpage extends BaseDialog<AddSubpageModel> {
-                                  entitySet: targetEntitySet,
-                                  navigationProperty: navProperty
-                              }
-                          }
-                      }
-                  }
-                : {
-                      changeType: 'appdescr_fe_addNewPage',
-                      parameters: {
-                          sourcePage: {
-                              id: this.runtimeControl.getId(),
-                              navigationSource: targetEntitySet
-                          },
-                          targetPage: {
-                              type: 'Component',
-                              id: `${targetEntitySet}ObjectPage`,
-                              name: 'sap.fe.templates.ObjectPage',
-                              routePattern: `${targetEntitySet}({key}):?query:`,
-                              settings: {
-                                  contextPath: `/${targetEntitySet}`,
-                                  editableHeaderContent: false
-                              }
-                          }
-                      }
-                  };
+        const entitySet = this.model.getProperty('/currentEntitySet');
+
+        let modifiedValue;
+        if (appType === 'fe-v2') {
+            modifiedValue = {
+                changeType: 'appdescr_ui_generic_app_addNewObjectPage',
+                reference,
+                parameters: {
+                    parentPage: {
+                        component: pageType,
+                        entitySet
+                    },
+                    childPage: {
+                        id: `ObjectPage|${navProperty}`,
+                        definition: {
+                            entitySet: targetEntitySet,
+                            navigationProperty: navProperty
+                        }
+                    }
+                }
+            };
+        } else {
+            const routePattern = generateRoutePattern(
+                this.model.getProperty('/routePattern'),
+                navProperty,
+                targetEntitySet
+            );
+            modifiedValue = {
+                changeType: 'appdescr_fe_addNewPage',
+                reference,
+                parameters: {
+                    sourcePage: {
+                        id: pageId,
+                        navigationSource: navProperty
+                    },
+                    targetPage: {
+                        type: 'Component',
+                        id: `${targetEntitySet}ObjectPage`,
+                        name: 'sap.fe.templates.ObjectPage',
+                        routePattern,
+                        settings: {
+                            contextPath: `/${targetEntitySet}`,
+                            editableHeaderContent: false,
+                            entitySet: targetEntitySet,
+                            pageLayout: '',
+                            controlConfiguration: {}
```

Comment:
Are these empty values required?

Reply:
Yes, framework change validation is complaining if they are not provided.